language-support/ts/codegen/src/TsCodeGenMain.hs

```diff
@@ -565,6 +567,52 @@ writePackageIdTs dir pkgId =
     T.writeFileUtf8 (dir </> "packageId.ts") $ T.unlines
       ["export default '" <> unPackageId pkgId <> "';"]
 
+writeIndexTs :: FilePath -> [Module] -> IO ()
+writeIndexTs packageSrcDir modules =
+  T.writeFileUtf8 (packageSrcDir </> "index.ts") lines
+  where
+    lines :: T.Text
+    lines = T.unlines $ header <> concatMap entry modules
+
+    header :: [T.Text]
+    header =
+      [ "import {default as __packageId} from \"./packageId\""
+      , "export const packageId = __packageId"
+      ]
+
+    entry :: Module -> [T.Text]
+    entry mod
+      | null $ defDataTypes mod = []
+      | otherwise = importDecl var path <> exportDecl var parts
+      where
+        parts = unModuleName (moduleName mod)
+        var = "__" <> T.intercalate "_" parts
```

Comment:
You already have the same logic for turning a module name into a file name in `genModuleRef`. Let’s factor this out into a function.

Reply:
ok

Reply:
Found further uses for this function in https://github.com/digital-asset/daml/pull/4951.